### PR TITLE
Out of gas

### PR DESCRIPTION
**Pull request by bitwiseguy**
Refactored code for "out-of-gas" error reporting.
Added -N flag to allow user to specify Nonce.

### CODE REVIEW

File: cmd/kaleido-go.go [hunk 1]
```diff
@@ -28,13 +28,11 @@ func initLogging(debugLevel int) {
 	switch debugLevel {
 	case 0:
 		log.SetLevel(log.ErrorLevel)
-		break
 	case 1:
 		log.SetLevel(log.InfoLevel)
-		break
 	default:
+		debugLevel = 2
```
Comment (peterbroadhurst) [inline cmd/kaleido-go.go:34]: I think the right default is `INFO` (and the default is set elsewhere int he `cobra config`), but if you think there are critical messages missing we should get those added at info-level

File: pkg/kldexerciser/kldworker.go [hunk 2]
```diff
@@ -383,25 +372,31 @@ func (w *Worker) Run() {
 			if err != nil {
 				w.error("TX send failed (%d/%d): %s", i, w.Exerciser.TxnsPerLoop, err)
 			} else {
-				w.Nonce++
 				txHashes = append(txHashes, txHash)
 			}
+			w.Nonce++
```
Comment (peterbroadhurst) [inline pkg/kldexerciser/kldworker.go]: I'm worried this is going to break the nonce logic.
I agree in the _special_ case of a failure with `status=1`, we need to bump the Nonce on a failure, but in the general case where transactions cannot be submitted if we bump the nonce we'll create gaps.

**Review by peterbroadhurst — CHANGES_REQUESTED** [review]
Thanks @sastokes23623 - really like the changes.
There are some minor comments above to address. The Nonce one is the most important I think.

File: cmd/kaleido-go.go (hunk 1 above)
Comment (bitwiseguy) [reply cmd/kaleido-go.go:34]: If the user does not specify the debug level (using -d), then the `cobra config` automatically sets it to '1', which is `INFO`.  This case-statement default just sets the debug level to `DEBUG` if the user provides any number other than 0 or 1.  Previously if they provided something like `-d 3`, the first printout to console would say `Debug level 3` even though there is no such thing.

File: pkg/kldexerciser/kldworker.go (hunk 2 above)
Comment (bitwiseguy) [reply pkg/kldexerciser/kldworker.go]: I will move this one back inside the `if else` block. I agree that the issue I was trying to address was a very specific case that does not need to be addressed if it risks the logic for the broader, more likely scenarios.

**Review by peterbroadhurst — APPROVED** [review]
👍

**Pull request merged**